Add Adjust SDK config plugin

Kerumen opened the pull request:
# Why

- Add config plugin for [Adjust SDK](https://github.com/adjust/react_native_sdk)
- Resolve https://github.com/expo/config-plugins/issues/35

# Test Plan

- Run `expo prebuild`
- On iOS, ensure the [5 libraries](https://github.com/adjust/react_native_sdk#ios-frameworks) are linked properly with the optional status:

![Screenshot 2022-01-12 at 10 07 11](https://user-images.githubusercontent.com/5436545/149098197-70a469ef-95a0-43ad-bc5a-e6bd7370947c.png)

- On Android, ensure [the permission](https://github.com/adjust/react_native_sdk#android-permissions) is here and [the dependencies](https://github.com/adjust/react_native_sdk#google-play-services) are added in the app `build.gradle` file:

![Screenshot 2022-01-12 at 10 10 11](https://user-images.githubusercontent.com/5436545/149098221-9e20c66a-34d9-4bb9-8dbc-de0a6daf5093.png)
![Screenshot 2022-01-12 at 10 10 43](https://user-images.githubusercontent.com/5436545/149098217-806ad972-1059-48c8-a25a-f54b0b9565c6.png)

EvanBacon commented on packages/react-native-adjust/src/withReactNativeAdjust.ts:
```diff
@@ -0,0 +1,112 @@
+};
+
+/**
+ * Apply react-native-adjust configuration for Expo SDK 42 projects.
```
```suggestion
 * Apply react-native-adjust configuration for Expo SDK +44 projects.
```

EvanBacon reviewed (commented):
Looks good so far, please:

- Add a "test plan" to the PR description.
- Include in the `apps/app` app.json and package.json
- Create a _very_ minimal example in `apps/app/src` so I can give it a try when upgrading SDK versions.

Kerumen commented:
@EvanBacon I updated the PR and the description. To be noted, I can't add the config-plugin in the example app from npm because it's not yet published so i added it with a `file:` reference.

EvanBacon commented on apps/app/package.json:
```diff
@@ -25,6 +25,7 @@
     "@config-plugins/detox": "^1.1.0",
     "@config-plugins/ffmpeg-kit-react-native": "^0.0.1",
     "@config-plugins/ios-stickers": "^0.0.4",
+    "@config-plugins/react-native-adjust": "file:../../packages/react-native-adjust",
```
```suggestion
    "@config-plugins/react-native-adjust": "0.0.1",
```

This should work as expected since the project is setup as a yarn workspace (published or not). Most of the magic takes place in the root `/package.json` if you're curious.

Kerumen replied:
Indeed, my bad!

EvanBacon commented on packages/react-native-adjust/src/withReactNativeAdjust.ts:
```diff
@@ -0,0 +1,112 @@
+  });
+
+  config = AndroidConfig.Permissions.withPermissions(config, [
+    "com.google.android.gms.permission.AD_ID",
```
According to the docs, this is for Android 12 (API 31), Expo currently supports API 30 so you may need to add a plugin to [bump the version](https://github.com/revtel/react-native-nfc-manager/pull/478).

Kerumen replied:
Ha yes. The docs says "For apps with target API level set to 30 (Android 11) or older, this permission is not needed.".

Don't you think it's the responsibility of the client application to update their target version?

EvanBacon replied:
I do but we don't really have a mechanism built in for this yet. The Expo SDK version is tied to a specific iOS and Android API version. Ideally every package would support the same versions and defensively code around versioning issues, but since this isn't always the case we need mechanisms for packages to ensure the runtime is suitable. 

Perhaps a config plugin property could be used to conditionally add the permission and bump the Android API version?

Kerumen replied:
That's a neat idea. I'll implement that!

Kerumen commented:
@EvanBacon I've added the option. I named it `targetAndroid12` because the Adjust docs says that the Google Advertising Id should be used since 2014. Hence, this could be confusing to name it `enableGoogleAdId`. Maybe `enableGoogleAdIdForAndroid12` but that's a bit long. Tell me if you have a better name in mind!

![Screenshot 2022-01-14 at 17 16 48](https://user-images.githubusercontent.com/5436545/149548629-5fffdd38-1a0b-43af-b148-65870e56a7a8.png)

EvanBacon commented:
Congrats `@config-plugins/react-native-adjust@0.0.1` is now published! Thanks for the hard work 🎉
